### PR TITLE
Package only from root plus powershell

### LINKED ISSUE #4492: Official mac build should not contain alias in the path
Attached to powershell `v6.0.0-beta.4` from VSCode and got this error
image

Notices the file path contains PII.

> When creating the packages, please ensure that the file path does not contain user names. That is, clone to /PowerShell on Unix, and C:\PowerShell for Windows. The debug symbols include the absolute path to the sources when built, so it should appear /PowerShell/src/powershell/System.Management.Automation, not /home/username/src/PowerShell/....

https://github.com/PowerShell/PowerShell/blob/master/docs/maintainers/releasing.md#building-packages

The package build script should check such problems and prevent from doing an official release from home directory.

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value                                                                                                   
----                           -----                                                                                                   
PSVersion                      6.0.0-beta                                                                                              
PSEdition                      Core                                                                                                    
GitCommitId                    v6.0.0-beta.4                                                                                           
OS                             Darwin 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEA...
Platform                       Unix                                                                                                    
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                 
PSRemotingProtocolVersion      2.3                                                                                                     
SerializationVersion           1.1.0.1                                                                                                 
WSManStackVersion              3.0   
```

### PR DESCRIPTION
**Pull request by TravisEz13**
fixes #4492

### CODE REVIEW

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -81,6 +83,14 @@ function Start-PSPackage {
         throw "Please ensure you have run 'Start-PSBuild $params'!"
     }
 
+    if($SkipReleaseChecks.IsPresent) {
+        Write-Warning "Skipping release checks."
+    }
+
+    if(!$Script:Options.RootInfo.IsValid  -and -not $SkipReleaseChecks.IsPresent){
```
Comment (daxian-dbw) [inline tools/packaging/packaging.psm1]: an `elseif` is better here.
```
elseif (!$Script:Options.RootInfo.IsValid) {
    throw $Script:Options.RootInfo.Warning
}
```

File: build.psm1 [hunk 2]
```diff
@@ -723,7 +723,24 @@ function New-PSOptions {
...
+    $RootInfo = @{RepoPath = $PSScriptRoot}
+
+    # the valid root is the root of the filesystem and the folder PowerShell
+    $RootInfo += @{ValidPath = Join-Path -Path ([system.io.path]::GetPathRoot($RootInfo.RepoPath)) -ChildPath 'PowerShell' }
+    
+    if($RepoInfo.RepoPath -ne $RootInfo.ValidPath)
+    {
+        $RootInfo += @{Warning = "Please ensure you repo is at the root of the file system and named 'PowerShell' (example: '$($RootInfo.ValidPath)'), when building and packaging for release!" }
+        $RootInfo += @{IsValid = $false}
+        Write-Warning -Message $RootInfo.Warning
```
Comment (daxian-dbw) [inline build.psm1]: Don't write the warning in `New-PSOptions`. We only care about the root path when it comes to a release build.

File: tools/travis.ps1 [hunk 3]
```diff
@@ -179,8 +179,9 @@ else
             $packageParams += @{Version=$version}
         }
         # Only build packages for branches, not pull requests
-        $packages = @(Start-PSPackage @packageParams)
-        $packages += Start-PSPackage  @packageParams -Type AppImage
+        $packages = @(Start-PSPackage @packageParams  -SkipReleaseChecks)
+        # Packaging AppImage depends on the deb package
+        $packages += Start-PSPackage  @packageParams -Type AppImage  -SkipReleaseChecks
```
Comment (daxian-dbw) [inline tools/travis.ps1]: There is an extra space before `-SkipReleaseChecks`, in both line 182 and 184

File: tools/packaging/packaging.psm1 (hunk 1 above)
Comment (TravisEz13) [inline tools/packaging/packaging.psm1]: Resolved

File: tools/travis.ps1 (hunk 3 above)
Comment (TravisEz13) [inline tools/travis.ps1]: Resolved

File: build.psm1 (hunk 2 above)
Comment (TravisEz13) [inline build.psm1]: Resolved

**Review by daxian-dbw — APPROVED** [review]

File: build.psm1 [hunk 4]
```diff
@@ -723,7 +723,23 @@ function New-PSOptions {
         $Top = [IO.Path]::Combine($PSScriptRoot, "src", "System.Management.Automation")
     }
 
-    return @{ Top = $Top;
+    $RootInfo = @{RepoPath = $PSScriptRoot}
+
+    # the valid root is the root of the filesystem and the folder PowerShell
+    $RootInfo += @{ValidPath = Join-Path -Path ([system.io.path]::GetPathRoot($RootInfo.RepoPath)) -ChildPath 'PowerShell' }
```
Comment (daxian-dbw) [inline build.psm1]: A minor comment about the way you use Hashtable: in this scenario it's better to write script like 
```
$RootInfo['ValidPath'] = Join-Path -Path ([system.io.path]::GetPathRoot($RootInfo.RepoPath)) -ChildPath 'PowerShell'
$RootInfo['IsValid'] = $true
$RootInfo['Warning'] = "Please ensure you repo is at the root of the file system and named 'PowerShell' (example: '$($RootInfo.ValidPath)'), when building and packaging for release!"
```
It's much less expensive than adding two hashtables.

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**